docs/user-guide/config-best-practices.md

```diff
@@ -0,0 +1,47 @@
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+![WARNING](http://kubernetes.io/img/warning.png)
+![WARNING](http://kubernetes.io/img/warning.png)
+![WARNING](http://kubernetes.io/img/warning.png)
+
+<h1>PLEASE NOTE: This document applies to the HEAD of the source
+tree only. If you are using a released version of Kubernetes, you almost
+certainly want the docs that go with that version.</h1>
+
+<strong>Documentation for specific releases can be found at
+[releases.k8s.io](http://releases.k8s.io).</strong>
+
+![WARNING](http://kubernetes.io/img/warning.png)
+![WARNING](http://kubernetes.io/img/warning.png)
+![WARNING](http://kubernetes.io/img/warning.png)
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+# Tips and tricks when working with config
+
+This document is meant to highlight and consolidate in one place configuration best practices that are introduced throughout the user-guide and getting-started documentation. This is a living document so if you think of something that is not on this list but might be useful to others, please don't hesitate to file an issue or submit a PR.
+
+1. When writing configuration, use the latest stable API version (currently v1). 
+1. Use YAML rather than JSON. They can be used interchangeably in almost all scenarios but YAML tends to be more user friendly for config.
+1. Group related objects together in a single file. This currently requires a List object in YAML, but it's still better than separate files.
+1. Use `kubectl create -f <directory>` where possible. This look for config objects in all `.yaml`, `.yml`, and `.json` files in `<directory>` and attempt to create them.
+1. Create a service before corresponding replication controllers so that the scheduler can spread the pods comprising the service. #11144 (or, create the RC without specifying replicas, create the service, then scale up the RC, which may work better in an example using progressive disclosure and may have benefits in real scenarios also, such as ensuring one replica works before creating lots of them)
+1. Don't use `hostPort` unless absolutely necessary (e.g., for a node daemon) as it will prevent certain scheduling configurations due to port conflicts. Use the apiserver proxying or port forwarding for debug/admin access, or a service for external service access. If you need to expose a pod's port on the host machine, consider using a [NodePort](services.md#type--loadbalancer) service before resorting to `hostPort`.
+1. Don't use `hostNetwork` for the same reasons as `hostPort`.
+1. Don't specify default values unnecessarily to simplify and minimize configs. Now that we default (e.g. the selector and labels in ReplicationController), configs should feel slimmer.
+1. Service should use a subset of the labels used by the replication controllers whose pods it targets. This will allow you to update replication controllers without updating the services that target them and will make rolling-update more effective.
```

Comment:
Link to managing-deployments.md#using-labels-effectively

I'd say:

Instead of attaching one label to a set of pods to represent a service (e.g., `service: myservice`) and another to represent the replication controller managing the pods (e.g., `controller: mycontroller`), attach labels that identify semantic attributes of your application or deployment and select the appropriate subsets in your service and replication controller, such as `{ app: myapp, tier: frontend, deployment: v3 }`. A service can be made to span multiple deployments, such as across rolling updates, by simply omitting release-specific labels from its selector, rather than updating a service's selector to match the replication controller's selector fully.